coordinatorlib/src/main/scala/com/socrata/datacoordinator/secondary/PlaybackToSecondary.scala

```diff
@@ -256,109 +293,228 @@ class PlaybackToSecondary[CT, CV](u: PlaybackToSecondary.SuperUniverse[CT, CV],
 
         val rs = new ResourceScope
 
-        val (effectiveStartingDataVersion, fakeRCP) =
-          (startingDataVersion to endingDataVersion).foldLeft((startingDataVersion, Delogger.RowsChangedPreview(0, 0, 0, false))) { (acc, i) =>
-            val (currentStartingDataVersion, rcp) = acc
+        sealed abstract class State
+        case object Unknown extends State
+        case class ConsolidatingRows(preview: Delogger.RowsChangedPreview) extends State
+        case object ConsolidatingRollups extends State
+
+        val (effectiveStartingDataVersion, state) =
+          (startingDataVersion to endingDataVersion).foldLeft((startingDataVersion, Unknown:State)) { (acc, i) =>
+            val (currentStartingDataVersion, state) = acc
+
+            def rcpOfState() =
+              state match {
+                case Unknown => Delogger.RowsChangedPreview(0, 0, 0, false)
+                case ConsolidatingRows(rcp) => rcp
+                case ConsolidatingRollups => throw new ResyncSecondaryException("Was consolidating rows, now rollups?")
```

Comment:
ok, this is most likely just another point of misunderstanding for me, but is the text here in reverse? (ie: should it actually be "Was consolidating rollups, now rows?"

Because it seems like we'll hit this case if the "state" from the previous accumulation is ConsolidatingRollups (meaning we *were* dealing with rollups stuff), but we called this function because we got a row-related event from the delogger?

Reply:
Hm yeah, I think you're right